### PR TITLE
CHI-418 - Converted Referrals and Notes to Customization Framework

### PR DESCRIPTION
**Pull request by andvirga**
Jira related:
https://bugs.benetech.org/browse/CHI-418

Primary reviewer: @GPaoloni 

Changes description:
-  Converted Referrals and Notes to use the new Customization Framework.
- Added `placeholder` prop to `textarea`.
- Keep backward compatibility with previous cases.

### CODE REVIEW

File: plugin-hrm-form/src/types/types.ts [hunk 1]
```diff
@@ -23,6 +25,14 @@ export type Incident = { [key: string]: string | boolean };
 
 export type IncidentEntry = { incident: Incident } & EntryInfo;
 
+export type Note = { [key: string]: string | boolean };
+
+export type NoteEntry = { note: string; counselor: string; date: string } & EntryInfo;
+
+export type Referral = { [key: string]: string | boolean };
+
+export type ReferralEntry = any;
```
Comment (nickhurlburt) [inline plugin-hrm-form/src/types/types.ts:34]: Is it necessary that this is `any`?  The other types here follow a more specific pattern.

Comment (nickhurlburt) [reply plugin-hrm-form/src/types/types.ts:34]: I can't make out if [this Insights code](https://github.com/tech-matters/flex-plugins/blob/master/plugin-hrm-form/src/services/InsightsService.ts#L178) will still do the right thing.

**Review by nickhurlburt — COMMENTED** [review]
Can you do a test for me?
1. Run this locally
2. Create a contact (offline or active are both fine)
3. Go find the task in the [Twilio Console](https://www.twilio.com/console/taskrouter/workspaces/WSc92e431ee05a5d0ac322f6c886c4aee2/tasks)
4. Create a referral on your contact and remember what you selected for "referred to"
5. Save the contact/case
6. Within five minutes, refresh the Twilio console view of your task
7. In the attributes field, confirm that `customer.customer_manager` is set to your "referred to" value.

This verifies that we're still sending [the right info](https://github.com/tech-matters/flex-plugins/blob/master/plugin-hrm-form/src/insightsConfig/zambia.ts#L159) to Insights

File: plugin-hrm-form/src/types/types.ts (hunk 1 above)
Comment (andvirga) [reply plugin-hrm-form/src/types/types.ts:34]: I did several tests and i'm not seeing the attribute "referredTo":
```json 
{
   "isContactlessTask":true,
   "helpline":"",
   "transferTargetType":"worker",
   "channelType":"default",
   "customers":{
      "zip":"",
      "customer_attribute_2":"",
      "area":"North-Western;Ikelenge",
      "customer_attribute_3":null,
      "customer_attribute_1":"Unknown",
      "gender":"Unknown",
      "city":"",
      "year_of_birth":"15",
      "acquisition_date":"Unknown",
      "customer_manager":"Hospital",
      "category":null,
      "region":"Unknown",
      "email":"Unknown"
   },
   "conversations":{
      "date":1611156780000,
      "communication_channel":"SMS",
      "conversation_attribute_5":"Unknown",
      "initiative":"",
      "conversation_attribute_6":null,
      "conversation_measure_5":null,
      "conversation_attribute_1":"Education",
      "conversation_attribute_2":"Child calling about self",
      "conversation_measure_2":null,
      "conversation_measure_1":null,
      "language":"Unknown",
      "conversation_measure_4":null,
      "conversation_measure_3":null,
      "conversation_attribute_7":null,
      "productive":null,
      "case":"1023"
   },
   "targetSid":"WK14efb75ca1ca5aa64866b2c5aca925dc"
}
```

Comment (nickhurlburt) [reply plugin-hrm-form/src/types/types.ts:34]: Great!  Then insights is still working correctly with the change. Thanks!

File: plugin-hrm-form/src/types/types.ts [hunk 2]
```diff
@@ -23,6 +25,14 @@ export type Incident = { [key: string]: string | boolean };
 
 export type IncidentEntry = { incident: Incident } & EntryInfo;
 
+export type Note = { [key: string]: string | boolean };
+
+export type NoteEntry = { note: string; counselor: string; date: string } & EntryInfo;
```
Comment (GPaoloni) [inline plugin-hrm-form/src/types/types.ts:30]: Note that `EntryInfo` is
`type EntryInfo = { createdAt: string; twilioWorkerId: string };`
and by looking at the code, I don't see where `createdAt` or `twilioWorkerId` are added to a note entry. It seems to me that this type is wrong.

**Review by GPaoloni — APPROVED** [review]
Everything looks good. The `AddNote` screen styles have changed a lot, maybe you can add `rows` property to the definition to make the textarea "bigger" (in vertical direction) but for having a wider textarea you can add more properties to the type definition that are later consumed by the form generator. I'm ok with the style tho, just mentioning this in case you are asked to change it.

**Comment by andvirga** [comment]
> Everything looks good. The `AddNote` screen styles have changed a lot, maybe you can add `rows` property to the definition to make the textarea "bigger" (in vertical direction) but for having a wider textarea you can add more properties to the type definition that are later consumed by the form generator. I'm ok with the style tho, just mentioning this in case you are asked to change it.

Thanks for your suggestions! I added some extra properties to `textarea` to meet the design with the mockups.

**Pull request merged**